resources/manager/factory_aws_test.go

```diff
@@ -0,0 +1,176 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package manager
+
+import (
+	"context"
+	"github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/utils/testhelper"
+	agentconfig "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type awsTestFetcherConfig struct {
+	fetching.AwsBaseFetcherConfig `config:",inline"`
+}
+
+type awsTestFetcher struct {
+	resourceCh chan fetching.ResourceInfo
+	cfg        awsTestFetcherConfig
+}
+
+func (f *awsTestFetcher) Fetch(ctx context.Context, cMetadata fetching.CycleMetadata) error {
+	f.resourceCh <- fetching.ResourceInfo{
+		Resource:      awsTestResource{AwsConfig: f.cfg.AwsConfig},
+		CycleMetadata: cMetadata,
+	}
+
+	return nil
+}
+
+func (f awsTestFetcher) Stop() {
+}
+
+type awsTestResource struct {
+	AwsConfig aws.ConfigAWS
+}
+
+func (a awsTestResource) GetMetadata() fetching.ResourceMetadata {
+	return fetching.ResourceMetadata{}
+}
+
+func (a awsTestResource) GetData() any {
+	return a.AwsConfig
+}
+
+func (a awsTestResource) GetElasticCommonData() any {
+	return nil
+}
+
+type awsTestFactory struct{}
+
+func (n *awsTestFactory) Create(log *logp.Logger, c *agentconfig.C, ch chan fetching.ResourceInfo) (fetching.Fetcher, error) {
+	cfg := awsTestFetcherConfig{}
+	err := c.Unpack(&cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &awsTestFetcher{ch, cfg}, nil
+}
+
+func awsMockedFetcherConfig(s *FactoriesTestSuite, awsConfig aws.ConfigAWS) *agentconfig.C {
+	c := agentconfig.NewConfig()
+	conf := config.DefaultConfig
+	conf.Type = config.InputTypeEKS
+	err := c.Merge(awsConfig)
+	s.Nil(err)
```

Comment:
```suggestion
	s.NoError(err)
```
please change it in other places too

Reply:
Fixed 💯